Source/WebCore/rendering/RenderTextControlSingleLine.cpp

```diff
@@ -152,8 +152,31 @@ void RenderTextControlSingleLine::layout()
         containerRenderer->layoutIfNeeded();
         oldContainerLogicalTop = containerRenderer->logicalTop();
         LayoutUnit containerLogicalHeight = containerRenderer->logicalHeight();
-        if (inputElement().hasAutoFillStrongPasswordButton() && innerTextRenderer && containerLogicalHeight != innerTextLogicalHeight) {
-            containerRenderer->mutableStyle().setLogicalHeight(Length { innerTextLogicalHeight, LengthType::Fixed });
+
+        auto autoFillStrongPasswordButtonSize = [&]() -> LayoutSize {
+            if (!inputElement().hasAutoFillStrongPasswordButton())
+                return { };
+
+            CheckedPtr autoFillButtonElement = inputElement().autoFillButtonElement();
+            if (!autoFillButtonElement)
+                return { };
+
+            CheckedPtr autoFillButtonRenderer = autoFillButtonElement->renderBox();
+            if (!autoFillButtonRenderer)
+                return { };
+
+            return autoFillButtonRenderer->logicalSize();
+        }();
+
+        if (!autoFillStrongPasswordButtonSize.isEmpty() && innerTextRenderer) {
+            auto newContainerHeight = innerTextLogicalHeight;
+            auto autoFillStrongPasswordButtonHeight = autoFillStrongPasswordButtonSize.height();
+
+            // Don't expand the container height if the AutoFill button is wrapped onto a new line.
+            if (containerLogicalHeight < innerTextLogicalHeight + autoFillStrongPasswordButtonHeight)
```

Comment:
@alanbaradlay Is there a better way to detect the case where the "Strong Password" button was flex wrapped onto it's own line?

Reply:
I don't see any straightforward way. This is all super custom layout :/ 
do you have an example when this happens?

Reply:
This happens whenever the `<input>`'s width is too small to fit both the text + "Strong Password" label. In the test cases in this PR, it can be observed by changing to width from 200px to 100px. Then, if we don't update the container height here, the container would be taller, and "Strong Label" would be visible on its own line.

The logic was introduced in https://commits.webkit.org/200350@main, with the goal of flex wrapping, and then hiding the "Strong Password" label.

Reply:
Another way to detect this scenario might be to check the relative x/y positions of the "Strong Password" box. If the y-position is greater than the input text, and the x-position is not greater than the max-x of the input text, I believe that would indicate a new line?

Not sure if that's more hacky...

Reply:
> The logic was introduced in https://commits.webkit.org/200350@main, with the goal of flex wrapping, and then hiding the "Strong Password" label.

I remember now. this is all awful.

Reply:
> Another way to detect this scenario might be to check the relative x/y positions of the "Strong Password" box. If the y-position is greater than the input text's max-y, and the x-position is not greater than the max-x of the input text, I believe that would indicate a new line?
> 
> Not sure if that's more hacky...
I actually prefer this since we are checking against wrapping.